### PR TITLE
Only creating gamepad data when gamepad information is available

### PR DESCRIPTION
**Pull request by rickfromwork**
Since the action profile's availability is separate from whether we use it, this change only creates a gamepad component on the input source if gamepad data is available.

### CODE REVIEW

File: Dependencies/xr/Source/OpenXR/XR.cpp [hunk 1]
```diff
@@ -1431,63 +1444,22 @@ namespace xr
...
-                    }
-
-                    // Get trackpad axes data
-                    float currentTrackpadXAxis, currentTrackpadYAxis;
-                    if (m_impl->QueryControllerVector2fAction(actionResources.ControllerGetTrackpadAxesAction, session, currentTrackpadXAxis, currentTrackpadYAxis))
+                    const auto& controllerInfo = sessionImpl.ControllerInfo;
+                    auto gamepadObject = InputSources[idx].GamepadObject;
+
+                    // Update gamepad data
+                    if ((m_impl->TryUpdateControllerFloatAction(actionResources.ControllerGetTriggerValueAction, session, gamepadObject.Buttons[controllerInfo.TRIGGER_BUTTON].Value)) &&
```
Comment (chrisfromwork) [inline Dependencies/xr/Source/OpenXR/XR.cpp:1452]: >TryUpdateControllerFloatAction [](start = 33, length = 30)

nit: can any of these fail while others succeed?

**Review by chrisfromwork — APPROVED** [review]
:shipit:

File: Dependencies/xr/Source/OpenXR/XR.cpp (hunk 1 above)
Comment (rickfromwork) [reply Dependencies/xr/Source/OpenXR/XR.cpp:1452]: I don't think so, but if any do fail I'd rather not send incomplete data (since babylonJS consumes all or none of it)

File: Dependencies/xr/Source/OpenXR/XR.cpp [hunk 2]
```diff
@@ -1431,63 +1444,22 @@ namespace xr
...
-                        inputSource.GamepadObject.Axes[controllerInfo.TRACKPAD_Y_AXIS] = currentTrackpadYAxis;
+                        // Only signal that gamepad data is available if the actions were available
+                        InputSources[idx].GamepadTrackedThisFrame = true;
                     }
-
-                    //Get trackpad click data
-                    bool currentTrackpadValue;
-                    if (m_impl->QueryControllerBooleanAction(actionResources.ControllerGetTrackpadClickAction, session, currentTrackpadValue))
-                    {
-                        inputSource.GamepadObject.Buttons[controllerInfo.TRACKPAD_BUTTON].Pressed = currentTrackpadValue;
```
Comment (satyapoojasama) [inline Dependencies/xr/Source/OpenXR/XR.cpp:1466]: why did you remove these? how are you updating the gamepad current values?

File: Dependencies/xr/Source/OpenXR/XR.cpp [hunk 3]
```diff
@@ -1115,53 +1113,68 @@ namespace xr
...
-           // query input action state
-           XrActionStateBoolean booleanActionState{XR_TYPE_ACTION_STATE_BOOLEAN};
-           XrActionStateGetInfo getInfo{XR_TYPE_ACTION_STATE_GET_INFO};
-           getInfo.action = controllerAction;
-           XrCheck(xrGetActionStateBoolean(session, &getInfo, &booleanActionState));
-           auto controllerInfo = sessionImpl.ControllerInfo;
-           if (booleanActionState.changedSinceLastSync)
-           {
+        // Returns true if the action is supported on the current input
+        bool TryUpdateControllerBooleanAction(XrAction controllerAction, XrSession session, bool& currentBooleanState)
```
Comment (satyapoojasama) [inline Dependencies/xr/Source/OpenXR/XR.cpp:1138]: I believe you mixed up two things here, action supported and changed since last frame are two different things. you might want probably a different function for action supported. It is a bit ambiguous

File: Dependencies/xr/Source/OpenXR/XR.cpp (hunk 2 above)
Comment (rickfromwork) [reply Dependencies/xr/Source/OpenXR/XR.cpp:1466]: I'm updating them inside of TryUpdateController*Action

Comment (satyapoojasama) [reply Dependencies/xr/Source/OpenXR/XR.cpp:1466]: okay, I just realized it :)

File: Dependencies/xr/Source/OpenXR/XR.cpp [hunk 4]
```diff
@@ -1431,63 +1444,22 @@ namespace xr
...
+                        (m_impl->TryUpdateControllerBooleanAction(actionResources.ControllerGetTrackpadTouchAction, session, gamepadObject.Buttons[controllerInfo.TRACKPAD_BUTTON].Touched)) &&
+                        (m_impl->TryUpdateControllerVector2fAction(actionResources.ControllerGetThumbstickAxesAction, session, gamepadObject.Axes[controllerInfo.THUMBSTICK_X_AXIS], gamepadObject.Axes[controllerInfo.THUMBSTICK_Y_AXIS])) &&
+                        (m_impl->TryUpdateControllerBooleanAction(actionResources.ControllerGetThumbstickClickAction, session, gamepadObject.Buttons[controllerInfo.THUMBSTICK_BUTTON].Pressed)))
                     {
-                        inputSource.GamepadObject.Axes[controllerInfo.TRACKPAD_X_AXIS] = currentTrackpadXAxis;
-                        inputSource.GamepadObject.Axes[controllerInfo.TRACKPAD_Y_AXIS] = currentTrackpadYAxis;
+                        // Only signal that gamepad data is available if the actions were available
+                        InputSources[idx].GamepadTrackedThisFrame = true;
```
Comment (rgerd) [inline Dependencies/xr/Source/OpenXR/XR.cpp:1461]: nit: Could set the `GamepadTrackedThisFrame` member directly to the result of the &'s instead of using an if, but I realize this may be a personal preference.

**Review by rgerd — APPROVED** [review]

**Review by satyapoojasama — APPROVED** [review]
